libgdf/src/types.cuh

```diff
@@ -0,0 +1,162 @@
+#ifndef GDF_CPPTYPES_H
+#define GDF_CPPTYPES_H
+
+#include <gdf/cffi/types.h>
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief  Wrapper structs for for the non-fundamental gdf_dtype types.
+ *
+ * These structs simply wrap a single member variable of a fundamental type 
+ * called "value". In order to access the underlying wrapped value, one can 
+ * use the "unwrap" function which will provide a reference to the underlying
+ * value.
+ *
+ * These wrapper structures are used in conjunction with the type_dispatcher to
+ * emulate "strong typedefs", i.e., provide opaque types that allow template 
+ * specialization. A normal C++ typedef is simply an alias and does not allow
+ * for specializing a template or overloading a function.
+ *
+ * The purpose of these "strong typedefs" is to allow distinguishing columns with
+ * different gdf_dtype types, but have the same underlying type. For example,
+ * the underlying type of both GDF_DATE32 and GDF_INT32 is int32_t. However, if
+ * one wished to specialize a functor invoked with the type_dispatcher to handle
+ * GDF_DATE32 different from GDF_INT32, that would not be possible with aliases.
+ *
+ * In the long term, these types should be updated such that they provide appropriate
+ * operators/member functions that implement the desired behavior for operating on
+ * elements of these types.
+ *
+ */
+/* ----------------------------------------------------------------------------*/
+namespace gdf
+{
+  /* --------------------------------------------------------------------------*/
+  /** 
+   * @brief  Wrapper struct for GDF_CATEGORY
```

Comment:
Instead of saying what GDF_DTYPE you're wrapping, considering putting an enum in there with that value. Always better to document with the code rather than a comment if you can express what you mean that way...